commit 6ae637a4120fc41676a1157c4b3f1c22a4376ad8 (refs/changes/52/2052/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-06-07T09:26:24-07:00 (2 years, 4 months ago)
    
    MON-362 test-amon failure: "could not determie the sdc-smartos image UUID"

diff --git a/test/prep.js b/test/prep.js
index 6744f93..0d79852 100644
--- a/test/prep.js
+++ b/test/prep.js
@@ -55,7 +55,7 @@ var ufdsClient;
 var amontestzone; // the test zone to use
 var headnodeUuid;
 var amonZoneUuid;
-var smartosImageUuid;
+var testImageUuid;
 var externalNetworkUuid;
 var testPackageUuid;
 var gzIp;
@@ -276,20 +276,18 @@ function getHeadnodeUuid(next) {
 }
 
 
-function getSmartosDatasetUuid(next) {
-    // No DSAPI in the DC yet, so hack it.
-    log('# Get "smartos" image UUID.');
-    exec('ls -1 /usbkey/datasets/sdc-smartos-*.*manifest | head -n1 '
-                    + '| xargs cat | json uuid',
-             function (err, stdout, stderr) {
+function getTestImageUuid(next) {
+    log('# Get a test image UUID (use the latest "amon" image origin)');
+    exec('sdc-imgadm list name=amon --latest -j | json 0.origin',
+            function (err, stdout, stderr) {
         if (err) {
             return next(err);
         }
-        smartosImageUuid = stdout.trim();
-        if (!smartosImageUuid) {
-            next(new Error('could not determie the sdc-smartos image UUID'));
+        testImageUuid = stdout.trim();
+        if (!testImageUuid) {
+            next(new Error('could not find a test image UUID'));
         }
-        log('# "smartos" dataset UUID is "%s".', smartosImageUuid);
+        log('# testImageUuid: %s', testImageUuid);
         next();
     });
 }
@@ -347,7 +345,7 @@ function createAmontestzone(next) {
         log('# Create a test zone for ulrich.');
         vmapiClient.createVm({
                 owner_uuid: ulrich.uuid,
-                image_uuid: smartosImageUuid,
+                image_uuid: testImageUuid,
                 server_uuid: headnodeUuid,
                 brand: 'joyent',
                 billing_id: testPackageUuid,
@@ -455,7 +453,7 @@ async.series([
         ldapClientUnbind,
         ufdsClientUnbind,
         getHeadnodeUuid,
-        getSmartosDatasetUuid,
+        getTestImageUuid,
         getExternalNetworkUuid,
         getTestPackageUuid,
         createAmontestzone,
